### PR TITLE
Smarter comparison for model inference output

### PR DESCRIPTION
**Pull request by lugi0**
Signed-off-by: Luca Giorgi <lgiorgi@redhat.com>

When the comparison is successful:
image

When an exception is thrown during comparison:
image

When the comparison of two elements fails:
image

### CODE REVIEW

**Review by jgarciao — DISMISSED** [review]
lgtm

File: libs/Helpers.py [hunk 1]
```diff
@@ -160,3 +160,42 @@ def uninstall_managed_starburst_using_addon_flow(self, cluster_name):
...
+            failures=[]
+            
+            def _inference_object_comparison(expected, received, threshold):
+                if isinstance(expected, dict):
+                    # if current element is a dict, compare the keys and then each value
+                    if not expected.keys()==received.keys(): failures.append([expected.keys(),received.keys()])
+                    for k in expected.keys():
+                        _inference_object_comparison(expected[k], received[k], threshold)
+                elif isinstance(expected, list):
+                    # if current element is a list, compare each value 1 by 1
```
Comment (bdattoma) [inline libs/Helpers.py:182]: you're using dictionary right? do you think this function will be used for list too?

Comment (lugi0) [reply libs/Helpers.py:182]: even though it starts from a dictionary, it then compares element by element, and the dictionary might contain lists

**Review by pablofelix — APPROVED** [review]
/lgtm

File: libs/Helpers.py (hunk 1 above)
Comment (bdattoma) [reply libs/Helpers.py:182]: oh I missed the line where you call the same function in the inside for loop. Okay

**Review by bdattoma — DISMISSED** [review]

**Review by pablofelix — APPROVED** [review]

File: tests/Tests/400__ods_dashboard/420__model_serving/420__model_serving.robot [hunk 2]
```diff
@@ -39,6 +39,11 @@ Verify Model Can Be Deployed Via UI
     Click Button    Create server
     Wait Until Page Contains    Data science projects
     Wait Until Page Contains Element    //button[.="Create data science project"]
+    ${already_present} =    Run Keyword And Return Status    Page Should Contain    ${PRJ_TITLE}
```
Comment (bdattoma) [inline tests/Tests/400__ods_dashboard/420__model_serving/420__model_serving.robot:42]: if you want there is the `Project Should Be Listed`, nothing will change from functionality pov

File: tests/Tests/400__ods_dashboard/420__model_serving/420__model_serving.robot [hunk 3]
```diff
@@ -39,6 +39,11 @@ Verify Model Can Be Deployed Via UI
     Click Button    Create server
     Wait Until Page Contains    Data science projects
     Wait Until Page Contains Element    //button[.="Create data science project"]
```
Comment (bdattoma) [inline tests/Tests/400__ods_dashboard/420__model_serving/420__model_serving.robot:41]: `Is Data Science Projects Page Open`

**Review by bdattoma — APPROVED** [review]
left some comments, nothing critical.

File: tests/Tests/400__ods_dashboard/420__model_serving/420__model_serving.robot (hunk 2 above)
Comment (lugi0) [reply tests/Tests/400__ods_dashboard/420__model_serving/420__model_serving.robot:42]: ty, need to do another pass on this kw and will fix this up then

**Pull request merged**